### ISSUE TITLE
Only remove apache host definitions on first installation

### ISSUE THREAD

**Issue by jean-gui** [privacyidea/privacyidea#834]
Installing privacyidea-apache2 package (2.20.1-1xenial) removes all existing virtual hosts from sites-enabled before (re-)adding its own.

I believe the responsible is https://github.com/privacyidea/privacyidea/blob/master/deploy/debian-ubuntu/privacyidea-apache2.postinst#L98
This line should be more careful about what it removes.

# What did you try to do?

Installed privacyidea-apache2 package (2.20.1-1xenial)

# What outcome did you expect?

privacyidea-apache2 installed and unrelated files untouched

# What outcome did you experience?

privacyidea-apache2 installed fine but all other virtual hosts were removed from /etc/apache2/sites-enabled/.

# Configuration

* privacyIDEA Version: 2.20.1-1xenial

* Installation method: ubuntu packages

## more details: 

* OS: Debian Stretch

* Webserver: Apache

* Tokendatabase: mariadb

**Comment by cornelinux** [comment]
I know that some might not like this. But this is the way it was working for the last 3 years and it worked for most of the users.

Why?

The meta package ``privacyidea-apache2`` is installed, if the administrator wants to setup a privacyIDEA machine. I.e. privacyIDEA will not run as another service next to wordpress or owncloud or any application, but as a dedicated system.

It is also documented here:
http://privacyidea.readthedocs.io/en/latest/installation/ubuntu.html

Nevertheless, I a curious about your opinion.

**Comment by jean-gui** [comment]
OK, that explains things, although I don't see any good reasons why this package would need to remove all active vhosts. It seems to me that it does so only because it was easier. The fact that it works for most users doesn't mean that it's not worth making it work for all.

The problem is that by removing random files like that, the package is effectively breaking other Ubuntu/Debian packages or custom configuration which IMO is against the philosophy of packaging. A package shouldn't break parts of the system without at least giving an opportunity to review those breaks.

Making the package only manage its own vhost and leave the other ones alone seems like a much saner approach.

**Comment by cornelinux** [comment]
Unfortunately this does not work out in all cases, since you might end up with conflicting configurations. And thus it will not work out for _all_ users again.
You will especially have problems, if there is already a configured SSL vhost.

I am looking forward to your suggestion.

**Comment by jean-gui** [comment]
How about creating the symlink only if it doesn't exist ? That way other vhosts would only break once.

**Comment by cornelinux** [comment]
I am so sorry. I can not do this. I am to dumb.
I spent the last two hours with this debian crap.

The problem is there is no way to test, if the file exists. A nice command like

   a2query -s privacyidea

will break the comple postinst script, if it returns a non-zero return code.
Testing for the existance of the link does also not work due to whoever knows internal maintainer script crap.

I am really sorry - I can not implment this. Anyway - I am open for pull requests.

**Comment by jean-gui** [comment]
Can't you do a simple `if [ ! -h /etc/apache2/site-enabled/privacyidea.conf ]` ?

**Comment by cornelinux** [comment]
I did use -e and -L. Which worked great at the command line.
But for some wired reason not in the postinst script! 

The additional strange thing is, that I am using the same mechanism to generate so other files. But something must be different here. :-/

And honestly then I totally lost any fun in investigating this... 
...then I overreacted and closed the ticket ;-)

I reopen it as know issue with low priority. Well... issue... actually it works as intented!
I will change the subject to reflect that this is not a bug ;-)
